mvrx/src/main/kotlin/com/airbnb/mvrx/PersistState.kt

```diff
@@ -31,30 +28,31 @@ annotation class PersistState
  * Iterates through all member properties annotated with [PersistState] and parcels them into a bundle that can be
  * saved with savedInstanceState.
  */
-internal fun <T : Any> T.persistState(assertCollectionPersistability: Boolean = false): Bundle {
-    val klass = this::class
-    val persistStateArgs = klass.primaryConstructor
-            ?.parameters
-            ?.filter { it.annotations.any { it.annotationClass == PersistState::class } }
-    if (persistStateArgs?.isEmpty() != false) {
-        return Bundle()
-    }
+internal fun <T : MvRxState> T.persistState(validation: Boolean = false): Bundle {
+    val jvmClass = this::class.java
+    // Find the first constructor annotated with @PersistState or return.
+    val constructor = jvmClass.constructors.firstOrNull { it.parameterAnnotations.any { it.any { it is PersistState } } } ?: return Bundle()
+
+    val bundle = Bundle()
+    constructor.parameterAnnotations.forEachIndexed { i, p ->
+        if (!p.any { it is PersistState }) return@forEachIndexed
+        // For each parameter in the constructor, there is a componentN function becasuse state is a data class.
+        // We can rely on this to be true because the MvRxMutabilityHelpers asserts that the state class is a data class.
+        val getter = try {
+            jvmClass.getDeclaredMethod("component${i + 1}").also { it.isAccessible = true }
+        } catch (e: NoSuchMethodException) {
```

Comment:
Might be better to just iterate over the methods and find it yourself rather than a full try/catch?